precompiles/vesting/integration_test.go

```diff
@@ -409,367 +428,458 @@ var _ = Describe("Interacting with the vesting extension", Ordered, func() {
 				fundClawbackVestingCheck := passCheck.
 					WithExpEvents(vesting.EventTypeFundVestingAccount)
 
-				_, _, err := s.factory.CallContractAndCheckLogs(funder.Priv, txArgs, callArgs, fundClawbackVestingCheck)
+				res, _, err := s.factory.CallContractAndCheckLogs(funderKey.Priv, txArgs, callArgs, fundClawbackVestingCheck)
 				Expect(err).ToNot(HaveOccurred(), "error while calling the contract: %v", err)
 				Expect(s.network.NextBlock()).To(BeNil())
 
+				fees := gasPrice.MulRaw(res.GasUsed)
+
 				// Check the vesting account
 				//
 				// NOTE: The vesting account is created with the vesting periods only, since the lockup periods are empty.
 				// The lockup periods are defaulted to instant unlocking, i.e. period length = 0.
-				s.ExpectVestingAccount(vestingKey.Addr, instantPeriods, defaultPeriods)
+				s.ExpectVestingAccount(vestingAccKey.Addr, instantPeriods, defaultPeriods)
 
-				// check that tx signer's balance is reduced by the fees paid
-				txSenderFinalBal := s.app.BankKeeper.GetBalance(s.ctx, s.address.Bytes(), s.bondDenom)
-				Expect(txSenderFinalBal.Amount).To(Equal(txSenderInitialBal.Amount.Sub(fees)))
+				// Query balances after precompile call.
+				finalBalances := s.GetBondBalances(vestingAccKey.AccAddr, funderKey.AccAddr, sdk.AccAddress(vestingCallerAddr.Bytes()))
+				vestAccFinalBal, funderFinalBal, vestingCallerFinalBal := finalBalances[0], finalBalances[1], finalBalances[2]
 
-				// check the contract's balance was deducted to fund the vesting account
-				contractFinalBal := s.app.BankKeeper.GetBalance(s.ctx, contractAddr.Bytes(), s.bondDenom)
-				Expect(contractFinalBal.Amount).To(Equal(sdk.ZeroInt()))
-
-				vestCoinsAmt := math.NewIntFromBigInt(defaultPeriods[0].Amount[0].Amount)
-				vestAccFinalBal := s.app.BankKeeper.GetBalance(s.ctx, toAddr.Bytes(), s.bondDenom)
-				Expect(vestAccFinalBal.Amount).To(Equal(vestAccInitialBal.Amount.Add(vestCoinsAmt)))
+				// Check balances after precompile call.
+				Expect(vestAccFinalBal).To(Equal(vestAccInitialBal.Add(math.NewInt(defaultFundingAmount))))
+				Expect(funderFinalBal).To(Equal(funderInitialBal.Sub(math.NewInt(defaultFundingAmount)).Sub(fees)))
+				Expect(vestingCallerFinalBal).To(Equal(vestingCallerInitialBal))
 			})
 
-			It(fmt.Sprintf("contract that calls funder - should NOT fund the vesting account with a smart contract different than the contract that calls the precompile (%s)", callType.name), func() {
+			// TODO: not clear the scope of this test
```

Comment:
@GAtom22 what is the purpose of this test?

Reply:
this test should fail due to lack of authorization. Is testing the authorization flow
the test creates an authorization but for another contract.